neuropacks/nsds_nwb/discrete.py

```diff
@@ -0,0 +1,167 @@
+        self.intervals['frq'] = self.intervals['frq'].astype(float)
+        self.intervals['amp'] = self.intervals['amp'].astype(float)
+        self.unique_frequencies = np.unique(self.intervals['frq'])
+        self.unique_amplitudes = np.unique(self.intervals['amp'])
+
+    def get_design_matrix(self, encoding='label'):
+        """Create the trial design matrix.
+
+        Parameters
+        ----------
+        encoding : str
+            How the design matrix should be encoded.
+            Options are ['label', 'value', 'onehot']
+
+        Returns
+        -------
+        design : dataframe
+            Design matrix dataframe. Each row is a trial and each column is a stimulus parameter.
+        """
+        idxs = self.intervals['sb'] == 's'
+        self.n_trials = np.sum(idxs)
+        if encoding in ['label', 'onehot']:
+            frq_enc = skpre.LabelEncoder()
+            frq_enc.fit(self.unique_frequencies)
+            amp_enc = skpre.LabelEncoder()
+            amp_enc.fit(self.unique_amplitudes)
+            design = pd.DataFrame({'frequency': frq_enc.transform(self.intervals['frq'].loc[idxs]),
+                                   'amplitude': amp_enc.transform(self.intervals['amp'].loc[idxs])})
+            if encoding == 'onehot':
+                frq_enc1 = skpre.OneHotEncoder()
+                frq_enc1.fit(frq_enc.transform(self.unique_frequencies)[:, np.newaxis])
+                amp_enc1 = skpre.OneHotEncoder()
+                amp_enc1.fit(amp_enc.transform(self.unique_amplitudes)[:, np.newaxis])
+                frqs = frq_enc1.transform(design['frequency'][:, np.newaxis]).toarray().astype(int)
+                amps = amp_enc1.transform(design['amplitude'][:, np.newaxis]).toarray().astype(int)
+                design = pd.DataFrame({'frequency': frqs.tolist(),
+                                       'amplitude': amps.tolist()})
+        elif encoding == 'value':
+            design = pd.DataFrame({'frequency': self.intervals['frq'].loc[idxs],
+                                   'amplitude': self.intervals['amp'].loc[idxs]})
+        else:
+            raise ValueError(f"`encoding` must be one of ['label', 'value', 'onehot'] was {encoding}")
+        return design
+
+
+class WhiteNoise(DiscreteStimuli):
+    def __init__(self, nwb_path):
+        super().__init__(nwb_path)
+
+    def get_design_matrix(self, encoding='value'):
+        """Create the trial design matrix.
+
+        Parameters
+        ----------
+        encoding : str
+            How the design matrix should be encoded.
+            Currently the only supported option is 'value'.
+
+        Returns
+        -------
+        design : dataframe
+            Design matrix dataframe. Each row is a trial and each column is a stimulus parameter.
+        """
+        self.n_trials = np.sum(self.intervals['sb'] == 's')
+
+        if encoding == 'value':
+            design = pd.DataFrame({'amplitude': np.ones(self.n_trials)})
```

Comment:
@jthermiz  I renamed the encoding option from 'label' to 'value', but let me know if this doesn't look right!

Reply:
Looks good, thanks!